include/seqan3/search/detail/search_configurator.hpp

```diff
@@ -25,6 +33,46 @@ namespace seqan3::detail
  */
 class search_configurator
 {
+private:
+    /*!\brief Select the search result based on the configuration and the index type.
+     *
+     * \tparam search_configuration_t The type of the configuration.
+     * \tparam index_t The type of the index.
+     */
+    template <typename search_configuration_t, typename index_t>
+    struct select_search_result
+    {
+    private:
+        //!\brief The cursor type of the index.
+        using index_cursor_type = typename index_t::cursor_type;
+        //!\brief The size type of the index.
+        using index_size_type = typename index_t::size_type;
+
+    public:
+        //!\brief The result type depending on the output configuration.
+        using type = std::conditional_t<search_traits<search_configuration_t>::search_return_index_cursor,
+                                        search_result<size_t, index_cursor_type, empty_type, empty_type>,
+                                        search_result<size_t, empty_type, index_size_type, index_size_type>>;
+    };
+
+    /*!\brief Selects the search algorithm based on the index type.
+     *
+     * \tparam search_configuration_t The type of the configuration.
+     * \tparam index_t The type of the index.
+     * \tparam policies_t A template parameter pack over the policies to specify the behavior of the algorithm.
+     */
+    template <typename configuration_t, typename index_t, typename ...policies_t>
+    struct select_search_algorithm
+    {
+        //!\brief The selected algorithm type based on the index.
+        using type =
+            lazy_conditional_t<bi_fm_index_specialisation<index_t>,
+                               lazy<instantiate_t,
```

Comment:
what or who is `instantiate_t`?

Reply:
It is a deferred instantiation template of the given type and it's arguments. It is defined in `core/type_traits/lazy.hpp`

Reply:
Well, I kind of made it more complex then necessary
I removed the additional instantiate_t. Thanks for spotting 👍